src/core/ext/filters/fault_injection/service_config_parser.cc

```diff
@@ -122,14 +122,9 @@ ParseFaultInjectionPolicy(const Json::Array& policies_json_array,
       }
     }
     // Parse max_faults
-    if (ParseJsonObjectField(json_object, "maxFaults",
-                             &fault_injection_policy.max_faults,
-                             &sub_error_list, false)) {
-      if (fault_injection_policy.max_faults < 0) {
-        sub_error_list.push_back(GRPC_ERROR_CREATE_FROM_STATIC_STRING(
-            "field:maxFaults error:should be zero or positive"));
-      }
-    }
+    ParseJsonObjectField(json_object, "maxFaults",
```

Comment:
I actually dislike this warning because it proves terribly fragile since it encourages input validation to be removed that might be useful should the type ever change.

Minimally if we're doing this I'd ask for a `static_assert(std::is_unsigned<decltype(fault_injection_policy.max_faults)>::value, "maxFaults should be unsigned")`

Reply:
Done. GCC's `-Wextra` turns on `-Wtype-limits`, but clang's `-Wextra` does not. This was already a problem, but our CI didn't catch it. I have a hunch that we need to enable a gcc strict build somewhere in the CI test suite.

Reply:
We could instead replace the uint32_t variables with signed ints, I don't think the unsignedness here is for anything other than asserting nonzero values. https://google.github.io/styleguide/cppguide.html#Integer_Types

We'll still need gcc strict builds somewhere.

Reply:
The uint32's are tied to envoy protos. Changing them is likely not worth the effort. Do you object to merging this PR as is?

CC @lidizheng